front/src/store/index.js

```diff
@@ -50,11 +60,42 @@ export default new Vuex.Store({
     },
     logout ({ commit }) {
       commit('CLEAR_USER_DATA');
+    },
+    recoveryPasswordSendEmail ({ commit }, payload) {
+      return new Promise((resolve, reject) => {
+        axios.post(`${serverPath}api/ForgotPassword`,
+          { Email: payload.email }).then(resp => {
+          if (resp.status !== 200) {
+            reject(resp.data);
+          } else {
+            commit('SET_EMAIL_DATA', payload.email);
+            resolve();
+          }
+        });
+      });
+    },
+    recoveryPasswordSendPassword ({ commit, state }, passwordandtoken) {
+      return new Promise((resolve, reject) => {
+        axios.post(`${serverPath}api/ForgotPassword/ResetPassword`,
+          { Email: state.email, Password: passwordandtoken.password }, {
+            headers: { Authorization: 'Bearer ' + passwordandtoken.token }
+          }).then(resp => {
+          if (resp.data !== '') {
+            commit('CLEAR_EMAIL_DATA');
+            resolve();
+          } else {
+            reject(resp.data);
+          }
+        });
+      });
     }
   },
   getters: {
     isLoggedIn (state) {
       return !!state.user;
+    },
+    getEmail (state) {
+      return state.email;
     }
```

Comment:
we used modules in store to separate by different parts different logic. you also could create store module for recovery password as locale, trip and user (in pull request yet)